subgraphs/aave-forks/protocols/aave-v3/src/mapping.ts

```diff
@@ -629,195 +607,180 @@ function getAssetPriceFallback(
+  // see https://github.com/aave/aave-v3-core/blob/1e46f1cbb7ace08995cb4c8fa4e4ece96a243be3/contracts/protocol/libraries/configuration/ReserveConfiguration.sol#L491
+  // for how to decode configuration data to get _liquidationProtocolFee
+  const liquidationProtocolFeeMask =
+    "0xFFFFFFFFFFFFFFFFFFFFFF0000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF";
+  const reserveFactorStartBitPosition = 152 as u8;
+
+  const pool = LendingPoolContract.bind(poolAddress);
+  const poolConfigData = pool.getConfiguration(reserve).data;
+  const liquidationProtocolFee = decodeConfig(
+    poolConfigData,
+    liquidationProtocolFeeMask,
+    reserveFactorStartBitPosition
+  )
+    .toBigDecimal()
+    .div(exponentToBigDecimal(INT_FOUR));
+
+  log.info("[storeLiquidationProtocolFee]market {} liquidationProtocolFee={}", [
+    market.id.toHexString(),
+    liquidationProtocolFee.toString(),
+  ]);
+  market._liquidationProtocolFee = liquidationProtocolFee;
   market.save();
 }
 
-function sortRewardTokens(market: Market): void {
-  if (market.rewardTokens!.length <= 1) {
-    return;
-  }
+function decodeConfig(
+  storedData: BigInt,
+  maskStr: string,
+  startBitPosition: u8
+): BigInt {
+  // aave-v3 stores configuration in packed bits (ReserveConfiguration.sol)
+  // decoding them by applying a bit_not mask and right shift by startBitPosition
+  // see https://github.com/aave/aave-v3-core/blob/1e46f1cbb7ace08995cb4c8fa4e4ece96a243be3/contracts/protocol/libraries/configuration/ReserveConfiguration.sol#L491
+  // for how to decode configuration data to get _liquidationProtocolFee
+
+  const maskArray = new Uint8Array(32);
+  maskArray.set(Bytes.fromHexString(maskStr));
+  // BITWISE NOT
+  for (let i = 0; i < maskArray.length; i++) {
+    maskArray[i] = ~maskArray[i];
+  }
+  // reverse for little endian
+  const configMaskBigInt = BigInt.fromUnsignedBytes(
+    Bytes.fromUint8Array(maskArray.reverse())
+  );
 
-  const tokens = market.rewardTokens;
-  const emissions = market.rewardTokenEmissionsAmount;
-  const emissionsUSD = market.rewardTokenEmissionsUSD;
-  multiArraySort(tokens!, emissions!, emissionsUSD!);
+  const config = storedData
+    .bitAnd(configMaskBigInt)
+    .rightShift(startBitPosition);
 
-  market.rewardTokens = tokens;
-  market.rewardTokenEmissionsAmount = emissions;
-  market.rewardTokenEmissionsUSD = emissionsUSD;
-  market.save();
+  return config;
 }
 
-function multiArraySort(
-  ref: Array<string>,
-  arr1: Array<BigInt>,
-  arr2: Array<BigDecimal>
-): void {
-  if (ref.length != arr1.length || ref.length != arr2.length) {
-    // cannot sort
-    return;
-  }
-
-  const sorter: Array<Array<string>> = [];
-  for (let i = 0; i < ref.length; i++) {
-    sorter[i] = [ref[i], arr1[i].toString(), arr2[i].toString()];
-  }
-
-  sorter.sort(function (a: Array<string>, b: Array<string>): i32 {
-    if (a[0] < b[0]) {
-      return -1;
+function getIsIsolatedFlag(event: ethereum.Event): boolean {
```

Comment:
should this be in the src/helpers.ts file?

Reply:
moved.

Reply:
good point, resolving